tools/autograd/derivatives.yaml

```diff
@@ -2953,3 +2953,6 @@
 - name: _reshape_copy(Tensor self, SymInt[] size) -> Tensor
   self: grad.reshape_symint(self.sym_sizes())
   result: auto_linear
+
+- name: all_reduce(Tensor self, str reduceOp, str tag, int[] ranks, int stride) -> Tensor
```

Comment:
have you thought about how to handle `str reduceOp`?

i think it is OK to leave it as str instead of a new C++ type.

however, we need a way to translate between str/enum, and I didn't think about where to put that translator yet.

currently, i think we only need this in python (since inductor also calls into py api), so maybe we can keep it together with the pg mapping utils.

Reply:
I think it should be in the same place as the op implementation.